PsfLauncher/main.cpp

```diff
@@ -74,16 +65,17 @@ int launcher_main(PCWSTR args, int cmdShow) noexcept try
 
     // At least for now, configured launch paths are relative to the package root
     std::filesystem::path packageRoot = PSFQueryPackageRootPath();
+	auto currentDirectory = (packageRoot / dirStr);
+
+	PsfPowershellScriptRunner powershellScriptRunner(appConfig, currentDirectory);
+
+
 
     // Launch the starting PowerShell script if we are using one.
-    auto startScriptInformation = PSFQueryStartScriptInfo();
-    if (startScriptInformation)
-    {        
-        THROW_HR_IF_MSG(ERROR_NOT_SUPPORTED, !CheckIfPowershellIsInstalled(), "PowerShell is not installed.  Please install PowerShell to run scripts in PSF");
-        RunScript(*startScriptInformation, packageRoot, dirStr, cmdShow);
-    }
+	THROW_IF_FAILED(powershellScriptRunner.RunStartingScript());

```

Comment:
>THROW_IF_FAILED [](start = 1, length = 15)

Since powershellscriptrunner already throws, you can remove this and make the member functions void returns.

Reply:
True.  Fixed.

---
In reply to: [316811851](https://github.com/microsoft/MSIX-PackageSupportFramework/pull/79#discussion_r316811851) [](ancestors = 316811851)